platform/darwin/src/MGLStyleValue_Private.h

```diff
@@ -322,6 +176,75 @@ class MGLStyleValueTransformer {
+        // defaultValue => default
+        if (styleFunction.defaultValue) {
+            NSCAssert([styleFunction.defaultValue isKindOfClass:[MGLStyleConstantValue class]], @"Default value must be constant");
+            rawFunction[@"default"] = [(MGLStyleConstantValue<ObjCType> *)styleFunction.defaultValue rawValue];
+        }
+        
+        // interpolationBase => base
+        if (styleFunction.interpolationBase) {
+            // TODO: does this direct cast work?
+            rawFunction[@"base"] = (NSNumber *) styleFunction.interpolationBase;
+        }
+        
+        // stops
+        if ([styleFunction isKindOfClass:[MGLCameraStyleFunction class]]) {
+            // zoom-only function
+            __block NSMutableArray *stops = [[NSMutableArray alloc] init];
+            [styleFunction.stops enumerateKeysAndObjectsUsingBlock:^(NSNumber * _Nonnull zoomKey, MGLStyleConstantValue<ObjCType> * _Nonnull outputValue, BOOL * _Nonnull stop) {
+                NSArray *rawStop = @[zoomKey, [outputValue rawValue]];
+                [stops addObject:rawStop];
+            }];
+        } else if ([styleFunction isKindOfClass:[MGLSourceStyleFunction class]]) {
+            rawFunction[@"property"] = ((MGLSourceStyleFunction<ObjCType> *)styleFunction).attributeName;
+            // property-only function
+            __block NSMutableArray *stops = [[NSMutableArray alloc] init];
+            [styleFunction.stops enumerateKeysAndObjectsUsingBlock:^(NSObject * _Nonnull propertyKey, MGLStyleConstantValue<ObjCType> * _Nonnull outputValue, BOOL * _Nonnull stop) {
+                NSArray *rawStop = @[propertyKey, [outputValue rawValue]];
+                [stops addObject:rawStop];
+            }];
+        } else if ([styleFunction isKindOfClass:[MGLCompositeStyleFunction class]]) {
+            // zoom-and-property function
+            MGLCompositeStyleFunction<ObjCType> *compositeStyleFunction = (MGLCompositeStyleFunction<ObjCType> *)styleFunction;
+            rawFunction[@"property"] = compositeStyleFunction.attributeName;
+            
+            __block NSMutableArray *stops = [[NSMutableArray alloc] init];
+            [compositeStyleFunction.stops enumerateKeysAndObjectsUsingBlock:^(NSNumber * _Nonnull zoomKey, NSDictionary * _Nonnull stopValue, BOOL * _Nonnull stop) {
+                for (NSObject *valueKey in stopValue.allKeys) {
+                    NSDictionary *stopKey = @{
+                         @"zoom": zoomKey,
+                         @"value": valueKey
+                     };
+                    MGLStyleConstantValue<ObjCType> *outputValue = stopValue[valueKey];
+                    NSCAssert([outputValue isKindOfClass:[MGLStyleConstantValue<ObjCType> class]], @"Stop outputs should be MGLStyleConstantValues");
+                    NSArray *rawStop = [NSArray arrayWithObjects:stopKey, [outputValue rawValue]];
+                    [stops addObject:rawStop]; // TODO weird: why doesn't stops.addObject(rawStop) work?
```

Comment:
Ahh, I see.  I missed the fact that the dot notation was specifically for property access!

> forcing your poor code reviewer to read Lisp.

😂